app/Console/Commands/MakeFilamentActionCommand.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+namespace App\Console\Commands;
+
+use Illuminate\Console\Command;
+use Illuminate\Contracts\Filesystem\FileNotFoundException;
+use Illuminate\Support\Facades\File;
+use Illuminate\Support\Str;
+
+use function Laravel\Prompts\info;
+use function Laravel\Prompts\select;
+use function Laravel\Prompts\text;
+use function Laravel\Prompts\warning;
+
+class MakeFilamentActionCommand extends Command
+{
+    protected $signature = 'make:filament-action {name?}
+                            {--form : Generate a form component action}
+                            {--table : Generate a table component action}
+                            {--table-bulk : Generate a table bulk action}
+                            {--custom-component : Generate a custom component action}
+                            {--infolist : Generate an infolist component action}
+                            {--notification : Generate a notification action}
+                            {--global-search : Generate a global search result action}';
+
+    protected $description = 'Create a new Filament action class';
+
+    /**
+     * @var array<string, string>
+     */
+    protected array $actionTypes = [
+        'form' => 'Filament\\Forms\\Components\\Actions\\Action',
+        'table' => 'Filament\\Tables\\Actions\\Action',
+        'table-bulk' => 'Filament\\Tables\\Actions\\BulkAction',
+        'custom-component' => 'Filament\\Actions\\Action',
+        'infolist' => 'Filament\\Infolists\\Components\\Actions\\Action',
+        'notification' => 'Filament\\Notifications\\Actions\\Action',
+        'global-search' => 'Filament\\GlobalSearch\\Actions\\Action',
+    ];
+
+    /**
+     * @throws FileNotFoundException
+     */
+    public function handle(): void
+    {
+        $name = $this->getNameArgument();
+        $className = $this->getClassName($name);
+        $type = $this->getActionType();
+        $actionClass = $this->actionTypes[$type];
+        $path = $this->getFilePath($className, $type);
+
+        if ($this->fileExists($path)) {
+            warning("$className already exists.");
+
+            return;
+        }
+
+        $stubContent = $this->getStubContent();
+        $this->generateActionFile($path, $className, $actionClass, $stubContent);
+
+        info("$className created successfully at:");
+        info($path);
+    }
+
+    private function getNameArgument(): string
+    {
+        return $this->argument('name') ?? text(
+            label: 'Enter the name of the Filament Action',
+            placeholder: 'MyActionName',
+            required: 'Action name is required.',
+            hint: "We will append 'Action' to this name, so no need to include it."
+        );
+    }
+
+    private function getClassName(string $name): string
+    {
+        return Str::studly($name).'Action';
```

Comment:
In the FilamentPHP make resource, there's an example that checks for a suffix to prevent appending another one. For instance, if I create an action called HelloAction, it would generate a file named `HelloActionAction`. Implementing this check will prevent that issue. Additionally, the hint: hint: "We will append 'Action' to this name, so no need to include it." will no longer be necessary.

Reply:
Nice touch. This has been added.